crates/db-boot/src/x86_64/mod.rs

```diff
@@ -17,6 +17,21 @@ pub mod mpspec;
 /// MP Table configurations used for defining VM boot status.
 pub mod mptable;
 
+/// Guest boot parameters used for config guest information.
+mod bootparam;
+
+// This is a workaround to the Rust enforcement specifying that any implementation of a foreign
+// trait (in this case `ByteValued`) where:
+// *    the type that is implementing the trait is foreign or
+// *    all of the parameters being passed to the trait (if there are any) are also foreign
+// is prohibited.
+#[repr(transparent)]
+#[derive(Copy, Clone, Default)]
+pub(crate) struct BootParamsWrapper(bootparam::boot_params);
```

Comment:
It's `pub(crate)` but not used, is it by design and will be used later?

Reply:
`(crate)` removed